### PR TITLE
Mark as unread

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Links

https://github.com/GetStream/ios-issues-tracking/issues/285

### 🎯 Goal

This PR adds an option to mark as unread a chat starting from a particular message

### 📝 Summary

Now it is possible to select a particular message, and mark the channel as unread starting from that message.

### 🛠 Implementation

ChannelController exposes the first unread message id, and this one is used in the UI to show a banner (not part of this PR) to visually distinguish new messages

### 🧪 Manual Testing Notes

1. Open a channel
2. Long press on a message
3. Select "Mark As Unread"

Expected result:
- The channel should be marked as unread starting from that message

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChatUI/Appearance+Images.swift [hunk 1]
```diff
@@ -189,6 +189,14 @@ public extension Appearance {
 
         public var messageActionInlineReply: UIImage = loadImageSafely(with: "icn_inline_reply")
         public var messageActionThreadReply: UIImage = loadImageSafely(with: "icn_thread_reply")
+        public var messageActionMarkUnread: UIImage = {
+            if #available(iOS 13.0, *) {
+                return UIImage(systemName: "message.badge")!
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/Appearance+Images.swift:194]: can't we fallback to the `loadImageSafely` here as well, instead of force unwrapping?

**Review by martinmitrevski — COMMENTED** [review]
Looks great so far 👍  We miss the WS event handling and the UI part, anything else?

File: Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift [hunk 2]
```diff
@@ -434,6 +450,8 @@ open class ChatMessageListVC: _ViewController,
         cell.setDecoration(for: .header, decorationView: delegate?.chatMessageListVC(self, headerViewForMessage: message, at: indexPath))
         cell.setDecoration(for: .footer, decorationView: delegate?.chatMessageListVC(self, footerViewForMessage: message, at: indexPath))
 
+        // TODO: Add logic to show/hide separator based on `firstUnreadMessageId`
```
Comment (ipavlidakis) [inline Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift:453]: This logic should be in the ChatChannelVC, right?

File: Tests/StreamChatTests/Database/DTOs/ChannelReadDTO_Tests.swift [hunk 3]
```diff
@@ -272,7 +272,103 @@ final class ChannelReadDTO_Tests: XCTestCase {
...
+        database.writeSessionCounter = 0
+
+        // WHEN
+        try database.writeSynchronously { session in
+            session.markChannelAsUnread(for: cid, userId: userId, from: messageId)
+        }
+
+        // THEN
+        XCTAssertEqual(database.writeSessionCounter, 1)
+        XCTAssertNotNil(readDTO(cid: cid, userId: userId))
```
Comment (ipavlidakis) [inline Tests/StreamChatTests/Database/DTOs/ChannelReadDTO_Tests.swift:325]: Do we want to check here the `unreadCount` to ensure that nothing is being updated?

File: Tests/StreamChatTests/Repositories/ChannelRepository_Tests.swift [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+final class ChannelRepository_Tests: XCTestCase {
+    private var repository: ChannelRepository!
+    private var database: DatabaseContainer_Spy!
+    private var apiClient: APIClient_Spy!
+
+    override func setUp() {
+        let client = ChatClient.mock
+        database = client.mockDatabaseContainer
+        apiClient = client.mockAPIClient
+        repository = ChannelRepository(database: database, apiClient: apiClient)
```
Comment (ipavlidakis) [inline Tests/StreamChatTests/Repositories/ChannelRepository_Tests.swift:18]: nit: Missing `super.setUp()`

File: Tests/StreamChatTests/Repositories/ChannelRepository_Tests.swift [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    override func tearDown() {
+        super.tearDown()
```
Comment (ipavlidakis) [inline Tests/StreamChatTests/Repositories/ChannelRepository_Tests.swift:22]: nit: Move `super.tearDown()` after this class' property deallocation

**Comment by ipavlidakis** [comment]
It looks great 🔥🔥🔥

File: Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift (hunk 2 above)
Comment (polqf) [reply Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift:453]: After the rebase with develop, yeah! Did not see that your changes were applied 😄

**Review by ipavlidakis — APPROVED** [review]
LGTM!

**Comment by alteral** [comment]
Is [6.0% Duplication](https://sonarcloud.io/component_measures?id=GetStream_stream-chat-swift&pullRequest=2519&metric=new_duplicated_lines_density&view=list) some sort of false-positive thingy?

**Pull request merged**